carmenadapter/switch.go

```diff
@@ -0,0 +1,87 @@
+package carmenadapter
+
+import (
+	"fmt"
+	carmen "github.com/Fantom-foundation/Carmen/go/state"
+	"github.com/Fantom-foundation/lachesis-base/hash"
+	"github.com/ethereum/go-ethereum/common"
+	"github.com/ethereum/go-ethereum/core/state"
+	"github.com/ethereum/go-ethereum/core/state/snapshot"
+	"math/big"
+	"os"
+	"path/filepath"
+)
+
+var carmenState carmen.State
+var carmenStateDb carmen.StateDB
+
+func InitializeCarmen(impl string, datadir string) error {
```

Comment:
Maybe we should use a more generalized naming schema, treating the DB variants equally and keeping the possibility for other implementations open, like it was done in Aida.

If so, I would propose to name this package just "state" or something similar and rename `InitializeCarmen` to `InitializeState`, tasked with initializing whatever implementation variant has been selected. And the rest of the project should be unaware what DB is actually used -- as is already almost archived by this PR.